### PR TITLE
Redux adjustments for extensions

### PR DESCRIPTION
**Pull request by diegopinate**
Use StateManager to manage redux store instead of having one defined in the application. This solves issues with tentative extensions enhancing the app's redux store.

### CODE REVIEW

File: packages/modules/itwin-viewer-react/src/services/Initializer.ts [hunk 1]
```diff
@@ -11,17 +12,23 @@ import { UrlDiscoveryClient } from "@bentley/itwin-client";
...
 import { initRpc } from "../config/rpc";
-import store from "../store";
 import { IModelBackendOptions, ItwinViewerInitializerParams } from "../types";
 import { ai, trackEvent } from "./telemetry/TelemetryService";
 
 // initialize required iModel.js services
 class Initializer {
   private static _initialized: Promise<void>;
+  // State manager for extensions to have redux store
+  private static _appStateManager: StateManager | undefined;
```
Comment (kckst8) [inline packages/modules/itwin-viewer-react/src/services/Initializer.ts:31]: do we need this? It doesn't appear to be used outside of initializing a new StateManager. Can we just initialize and not store it?

**Comment by kckst8** [comment]
not sure what is up with the CI build. May need to rebase? We had some failed publishes that may have put rush change tracking into a strange state

File: packages/modules/itwin-viewer-react/src/services/Initializer.ts (hunk 1 above)
Comment (diegopinate) [reply packages/modules/itwin-viewer-react/src/services/Initializer.ts]: Yep, it felt weird to just initialize it without keeping it around. I don't like how the constructor sets the singleton. Removed the variable.

**Comment by diegopinate** [comment]
Rebased to see if it fixes the CI

**Pull request merged**